test/prefer-dataset.js

```diff
@@ -0,0 +1,25 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/prefer-dataset';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+ruleTester.run('prefer-dataset', rule, {
+	valid: [
+		'element.dataset.unicorn = \'🦄\';',
+		'element.dataset[\'unicorn\'] = \'🦄\';',
+		'element[setAttribute](\'data-unicorn\', \'🦄\');',
+		'element.setAttribute(\'foo\', \'bar\');'
+	],
+	invalid: [
+		{
+			code: 'element.setAttribute(\'data-unicorn\', \'🦄\');',
+			errors: [{message: 'Prefer `dataset` over `setAttribute`'}],
+			output: 'element.dataset.unicorn = \'🦄\';'
+		}
```

Comment:
A few more test cases worth checking:

* `element.setAttribute(\'data-foo-bar\', \'zaz\')`
* `element.setAttribute(\'data-🦄\', \'🦄\')`
* `element.setAttribute(foo, bar)`
* `element.setAttribute(\'data-foo2\', \'bar\')`
* `element.setAttribute(\'data-foo:bar\', \'zaz\')`
* `element.setAttribute(\'data-foo\', /* comment */ \'bar\')`
* ```
element.setAttribute(
  \'data-foo\', // comment
  \'bar\', // comment
)
```

Note that the fix conversions should follow the description in the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset):

> The name of a custom data attribute in JavaScript is the name of the same HTML attribute but in camelCase and with no dashes, dots, etc.

I think this means `element.setAttribute('data-date-of-birth', '1960-10-03')` should be fixed to `element.dataset.dateOfBirth = '1960-10-03'`.

Reply:
Thank you for your feedback :smiley: